tests/integration/commands/dev/dev.test.mjs

```diff
@@ -0,0 +1,479 @@
+// Handlers are meant to be async outside tests
+import fs from 'fs/promises'
+import { join } from 'path'
+
+import jwt from 'jsonwebtoken'
+import fetch from 'node-fetch'
+import { test, describe } from 'vitest'
+
+import { withDevServer } from '../../utils/dev-server.cjs'
+import { startExternalServer } from '../../utils/external-server.cjs'
+import got from '../../utils/got.cjs'
+import { withMockApi } from '../../utils/mock-api.cjs'
+import { withSiteBuilder } from '../../utils/site-builder.cjs'
+
+describe.concurrent('command/dev', () => {
+  test('should return 404.html if exists for non existing routes', async (t) => {
+    await withSiteBuilder('site-with-shadowing-404', async (builder) => {
+      builder.withContentFile({
+        path: '404.html',
+        content: '<h1>404 - Page not found</h1>',
+      })
+
+      await builder.buildAsync()
+
+      await withDevServer({ cwd: builder.directory }, async (server) => {
+        const response = await got(`${server.url}/non-existent`, { throwHttpErrors: false })
```

Comment:
There are a few remaining `got`s in this test file

Reply:
Done!